chia/server/server.py

```diff
@@ -158,14 +169,43 @@ def __init__(
             self.introducer_peers = IntroducerPeers()
 
         if self._local_type is not NodeType.INTRODUCER:
-            self._private_cert_path, self._private_key_path = private_ssl_paths(root_path, config)
+            _private_cert_path, _private_key_path = private_ssl_paths(root_path, config)
+        else:
+            _private_cert_path, _private_key_path = None, None
+
         if self._local_type is not NodeType.HARVESTER:
-            self.p2p_crt_path, self.p2p_key_path = public_ssl_paths(root_path, config)
+            p2p_crt_path, p2p_key_path = public_ssl_paths(root_path, config)
+        else:
+            p2p_crt_path, p2p_key_path = None, None
+
+        # If node has public cert use that one for id, if not use private.
+        self.node_id = calculate_node_id(p2p_crt_path or _private_cert_path)
```

Comment:
I feel like this could use some reorganization.  Make the conditions check what is rather than what isn't (also use `==` not `is`), use `if`/`elif`, and then accordingly avoid the `or` object selection at the end.  I personally dislike that form.

Reply:
Not really what you requested and im not sure where you would put an `elif` there, but i refactored all a bit more with the last force push, let me know!